src/utils.ts

```diff
@@ -54,18 +60,48 @@ export const getDataFolder = () => {
 	if (process.argv[2] === 'build') return 'dist';
 
 	let devPath: string | undefined;
-	let folder: string;
-
-	if (getOs() === 'WIN') folder = devPath || path.resolve(process.env.APPDATA!, 'BetterDiscord', 'themes');
-	else if (getOs() === 'MACOS')
-		folder = devPath || path.resolve(process.env.HOME!, 'Library', 'Application Support', 'BetterDiscord', 'themes');
-	else if (getOs() === 'LINUX') folder = devPath || path.resolve(process.env.HOME!, '.local', 'share', 'BetterDiscord', 'themes');
-	else throw new Error('Cannot determine your OS.');
-
-	if (!fs.existsSync(getPath(folder))) {
-		log.error(`Directory does not exist: ${log.code('`' + getPath(folder) + '`')}`);
+	var folder = '';
+
+	const checkDir = (dir: string, exitOnError: boolean) => {
+		if (!fs.existsSync(getPath(dir))) {
+			switch (exitOnError) {
+				case true:
+					log.error(`Directory does not exist: ${log.code('`' + getPath(dir) + '`')}`);
+				case false:
+					log.warning(`Directory does not exist: ${log.code('`' + getPath(dir) + '`')}`);
+			}
+		}
 	}
 
+	switch (String(getOs()))  {
+		case 'WIN':
+			folder = devPath || path.resolve(process.env.APPDATA!, 'BetterDiscord', 'themes');
+			checkDir(folder, true);
```

Comment:
Why have `checkDir` in each case if you're just going to set the `folder` variable?